src/sftp.c

```diff
@@ -1554,7 +1554,7 @@ static ssize_t sftp_read(LIBSSH2_SFTP_HANDLE * handle, char *buffer,
             if(chunk->lefttosend) {
                 /* if the chunk still has data left to send, we shouldn't wait
                    for an ACK for it just yet */
-                if(bytes_in_buffer > 0) {
+                if(bytes_in_buffer >= 0) {
```

Comment:
Pointed out by the CI build: `bytes_in_buffer` is a `size_t` (unsigned). It will *always* be larger than or equal to zero...

Reply:
Right, and looking at this more I wonder if bytes_in_buffer should be made a ssize_t and initialized at -1 or leave bytes_in_buffer as a size_t and remove the is it greater than 0 check.